spec.html

```diff
@@ -38589,6 +38768,281 @@ <h1>JSON [ @@toStringTag ]</h1>
+      <p>The FinalizationRegistry constructor:</p>
+      <ul>
+        <li>
+          has a [[Prototype]] internal slot whose value is the intrinsic object %Function.prototype%.
+        </li>
+        <li>has the following properties:</li>
+      </ul>
+
+      <emu-clause id="sec-finalization-registry.prototype">
+        <h1>FinalizationRegistry.prototype</h1>
+        <p>
+          The initial value of `FinalizationRegistry.prototype` is the intrinsic %FinalizationRegistry.prototype% object.
+        </p>
+        <p>
+          This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *false* }.
+        </p>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-properties-of-the-finalization-registry-prototype-object">
+      <h1>Properties of the FinalizationRegistry Prototype Object</h1>
+      <p>The FinalizationRegistry prototype object:</p>
+      <ul>
+        <li>is the intrinsic object <dfn>%FinalizationRegistry.prototype%</dfn>.</li>
+        <li>
+          has a [[Prototype]] internal slot whose value is %Object.prototype%.
+        </li>
+        <li>is an ordinary object.</li>
+        <li>
+          does not have [[Cells]] and [[CleanupCallback]] internal slots.
+        </li>
+      </ul>
+
+      <emu-clause id="sec-finalization-registry.prototype.constructor">
+        <h1>FinalizationRegistry.prototype.constructor</h1>
+        <p>The initial value of `FinalizationRegistry.prototype.constructor` is the intrinsic object %FinalizationRegistry%.</p>
+      </emu-clause>
+
+      <emu-clause id="sec-finalization-registry.prototype.register">
+        <h1>FinalizationRegistry.prototype.register ( _target_, _heldValue_ [ , _unregisterToken_ ] )</h1>
+        <p>The following steps are taken:</p>
+        <emu-alg>
+          1. Let _finalizationRegistry_ be the *this* value.
+          1. Perform ? RequireInternalSlot(_finalizationRegistry_, [[Cells]]).
+          1. If Type(_target_) is not Object, throw a *TypeError* exception.
+          1. If SameValue(_target_, _heldValue_), throw a *TypeError* exception.
+          1. If Type(_unregisterToken_) is not Object, then
+            1. If _unregisterToken_ is not *undefined*, throw a *TypeError* exception.
+            1. Set _unregisterToken_ to ~empty~.
+          1. Let _cell_ be the Record { [[WeakRefTarget]] : _target_, [[HeldValue]]: _heldValue_, [[UnregisterToken]]: _unregisterToken_ }.
+          1. Append _cell_ to _finalizationRegistry_.[[Cells]].
+          1. Return *undefined*.
+        </emu-alg>
+
+        <emu-note>
+          <p>
+          Based on the algorithms and definitions in this specification, across the time when _cell_ is in _finalizationRegistry_.[[Cells]], then _cell_.[[Holdings]] is live; however, _cell_.[[UnregisterToken]] and _cell_.[[Target]] are not necessarily live.  For example, registering an object with itself as its unregister token would not keep the object alive forever.
```

Comment:
`[[Holdings]]` does not appear to be a thing. I assume this should be `[[HeldValue]]`?

Also, nit: `across the time when _cell_ is in _finalizationRegistry_.[[Cells]], then _cell_.[[Holdings]] is live` does not read very clearly to me. Can I suggest

```suggestion
          Based on the algorithms and definitions in this specification, _cell_.[[HeldValue]] is live while _cell_ is in _finalizationRegistry_.[[Cells]]; however, this does not necessarily mean that _cell_.[[UnregisterToken]] or _cell_.[[Target]] are live.  For example, registering an object with itself as its unregister token would not keep the object alive forever.
```